streaming/src/main/scala/org/apache/spark/streaming/ui/StreamingJobProgressListener.scala

```diff
@@ -101,6 +101,7 @@ private[streaming] class StreamingJobProgressListener(ssc: StreamingContext)
     runningBatchUIData(batchStarted.batchInfo.batchTime) = BatchUIData(batchStarted.batchInfo)
     waitingBatchUIData.remove(batchStarted.batchInfo.batchTime)
 
+    // TODO. this should be fixed when input stream is not receiver based stream.
```

Comment:
All these TODOs means that: now `BatchInfo#streamIdtoNumRecords` will track both the input stream's received data (for receiver-based stream) and to-be-processed data number (for receiver-less stream). So the semantics in `StreamingJobProgressListener` is slightly different.

Take `totalReceivedRecords` as an example: should we add the receiver-less input stream's number of to-be-processed record into here, since it is not actually received yet. So we have to figure out this semantics difference.